Formula/uuu.rb

```diff
@@ -0,0 +1,28 @@
+class Uuu < Formula
+  desc "Universal Update Utility, mfgtools 3.0. NXP I.MX Chip image deploy tools"
+  homepage "https://github.com/NXPmicro/mfgtools"
+  url "https://github.com/NXPmicro/mfgtools/releases/download/uuu_1.4.139/uuu_source-1.4.139.tar.gz"
+  sha256 "25a24575f9e4781b78f8f0930f751ed371dc1a00fc22d543e9cbe4a9a70a615c"
+  license "BSD-3-Clause"
+
+  head "https://github.com/NXPmicro/mfgtools.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+
+  # Change to stable when libusb 1.0.25 is released
+  # https://github.com/libusb/libusb/pull/911
+  depends_on "libusb" => :head
+  depends_on "libzip"
+  depends_on "openssl@1.1"
+
+  def install
+    system "cmake", "-S", ".", "-Wno-dev", *std_cmake_args
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    assert_match "Universal Update Utility", shell_output("#{bin}/uuu -h")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.